internal/appfunction/context.go

```diff
@@ -280,44 +282,41 @@ func (appContext *Context) PipelineId() string {
 }
 
 // Publish pushes data to the MessageBus using configured topic
-func (appContext *Context) Publish(data any) error {
-	messageClient := bootstrapContainer.MessagingClientFrom(appContext.Dic.Get)
-	if messageClient == nil {
-		return fmt.Errorf(messageBusDisabledErr)
-	}
+func (appContext *Context) Publish(data any, contentType string) error {
+	topic := container.ConfigurationFrom(appContext.Dic.Get).Trigger.PublishTopic
 
-	triggerConfig := container.ConfigurationFrom(appContext.Dic.Get).Trigger
-	baseTopic := bootstrapContainer.ConfigurationFrom(appContext.Dic.Get).GetBootstrap().MessageBus.BaseTopicPrefix
-
-	payload, err := json.Marshal(data)
-	if err != nil {
-		return fmt.Errorf("failed to marshal data for publishing using configured topic: %v", err)
-	}
-	message := types.NewMessageEnvelope(payload, context.Background())
-	err = messageClient.Publish(message, common.BuildTopic(baseTopic, triggerConfig.PublishTopic))
+	err := appContext.PublishWithTopic(topic, data, contentType)
 	if err != nil {
-		return fmt.Errorf("%v: %v", publishDataErr, err)
+		return err
 	}
 	return nil
 }
 
-// Publish pushes data to the MessageBus for a given topic
-func (appContext *Context) PublishWithTopic(topic string, data any) error {
+// PublishWithTopic pushes data to the MessageBus for a given topic
+func (appContext *Context) PublishWithTopic(topic string, data any, contentType string) error {
 	messageClient := bootstrapContainer.MessagingClientFrom(appContext.Dic.Get)
 	if messageClient == nil {
 		return fmt.Errorf(messageBusDisabledErr)
 	}
 
 	config := bootstrapContainer.ConfigurationFrom(appContext.Dic.Get)
 
-	full_topic := common.BuildTopic(config.GetBootstrap().MessageBus.BaseTopicPrefix, topic)
+	formattedTopic, err := appContext.ApplyValues(topic)
+	if err != nil {
+		return fmt.Errorf("%v: %v", topicFormatErr, err)
```

Comment:
does err contain the value of topic?  if not, it would be helpful to add topic as part of err msg. if so, then we are fine.

Reply:
Yes, it does. :-)